crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -613,26 +634,23 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
                     .ok_or(ResolveError::Unregistered)?;
                 self.visited.insert(package_name.clone());
 
+                let empty_version_map = VersionMap::default();
                 let version_map = match *versions_response {
                     VersionsResponse::Found(ref version_map) => version_map,
-                    // Short-circuit if we do not find any versions for the package
```

Comment:
We don't short circuit anymore because it might be available in the installed packages.

Reply:
Does that come at a cost? I assume it's negligible?

Reply:
I think it's a small cost, but I think this wasn't an optimization really it was just a "we don't have a version map so we can't call select". We still short circuit if no versions are found locally.